libs/contrib/Network/Socket/Data.idr

```diff
@@ -0,0 +1,172 @@
+||| Low-Level C Sockets bindings for Idris. Used by higher-level, cleverer things.
+||| Types used by Network.Socket.Raw and Network.Socket.
+|||
+||| Original (C) SimonJF, MIT Licensed, 2014
+||| Modified (C) The Idris Community, 2015, 2016
+module Network.Socket.Data
+
+%access public export
+
+-- ------------------------------------------------------------ [ Type Aliases ]
+
+ByteLength : Type
+ByteLength = Int
+
+ResultCode : Type
+ResultCode = Int
+
+||| Protocol Number.
+|||
+||| Generally good enough to just set it to 0.
+ProtocolNumber : Type
+ProtocolNumber = Int
+
+||| SocketError: Error thrown by a socket operation
+SocketError : Type
+SocketError = Int
+
+||| SocketDescriptor: Native C Socket Descriptor
+SocketDescriptor : Type
+SocketDescriptor = Int
+
+Port : Type
+Port = Int
+
+-- --------------------------------------------------------------- [ Constants ]
+
+||| Backlog used within listen() call -- number of incoming calls
+BACKLOG : Int
+BACKLOG = 20
+
+EAGAIN : Int
+EAGAIN =
```

Comment:
EAGAIN is a constant defined in errno.h. So this is probably fine (unsafePerformIO is currently the only way to get "pure" things out of foreign function calls).